WIP release-23.1: testserver: add unique message for test servers / clusters

yuzefovich opened the pull request:
Epic: None

yuzefovich committed:
testserver: add unique message for test servers / clusters

Every time we start a test cluster / server we'll now log "considering
starting a default test tenant" string.

yuzefovich committed:
WIP force external process tenant randomization

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115151)
<!-- Reviewable:end -->

yuzefovich committed:
testserver: WIP on more decision logs

Add "default test tenant disabled due to license check" and
"default test tenant disabled due to boolean flag".

Also "maybeStartDefaultTestTenant is called" to be in line with messages
about above decisions (which can happen multiple times per cluster).

yuzefovich committed:
WIP tweak ccl package to trigger CCL CI target
